internal/types/compare.go

```diff
@@ -254,3 +258,103 @@ func compareNumbers(a float64, b int64) CompareResult {
 
 	return CompareResult(bigA.Cmp(bigB))
 }
+
+// CompareArrays compares indices of a filter array according to indices of a document array;
+// returns Equal when a document array contains another array(subarray) that equals filter array.
+func CompareArrays(filterArr, docArr *Array) CompareResult {
+	if docArr.Len() == 0 && filterArr.Len() == 0 {
+		return Equal
+	}
+	if filterArr.Len() == 0 {
+		return Incomparable
+	}
+
+	EntireArrayResult, subArrayEquality := Incomparable, Incomparable
+
+	for i := 0; i < docArr.Len(); i++ {
+		arrValue := must.NotFail(docArr.Get(i))
+		switch arrValue := arrValue.(type) {
+		case *Array:
+			filterArrValue := must.NotFail(filterArr.Get(i))
+			switch filterArrValue := filterArrValue.(type) {
+			case *Array:
+				res := CompareArrays(filterArrValue, arrValue)
+				res = handleSubArrayComparingResult(i, &res, &EntireArrayResult, &subArrayEquality)
+				continue
+
+			default:
+				res := CompareArrays(filterArr, arrValue)
+				res = handleSubArrayComparingResult(i, &res, &EntireArrayResult, &subArrayEquality)
+			}
+			continue
+
+		// TODO: case Document
```

Comment:
should be here a link to an issue?

Reply:
Not nessesary. I'm going to work on this `TODO` as soon as this PR merged. I just want to make small and simple PRs. baby steps :smile: